virtual_ecosystem/models/abiotic/wind.py

```diff
@@ -128,685 +116,88 @@ def calculate_roughness_length_momentum(
-    )
-    stability_term = (
-        stable_temperature_gradient_intercept
-        * (1 + stability_factor * richardson_number) ** 0.5
-        + 2 * stable_wind_shear_slope * richardson_number
-        - stable_temperature_gradient_intercept
-    ) / (
-        2 * stable_wind_shear_slope * (1 - stable_wind_shear_slope * richardson_number)
-    )
-    sel = np.where(temperature_gradient <= 0)  # Unstable conditions
-    stability_term[sel] = richardson_number[sel]
-
-    # Initialize phi_m and phi_h with values for stable conditions
-    phi_m = 1 + (yasuda_stability_parameters[0] * stability_term) / (1 + stability_term)
-    phi_h = phi_m.copy()
-
-    # Adjust for unstable conditions
-    phi_m[sel] = 1 / (1 - yasuda_stability_parameters[2] * stability_term[sel]) ** 0.25
-    phi_h[sel] = phi_m[sel] ** 2
-
-    # Calculate mean values across the vertical axis for phi_m and phi_h
-    phi_m_mean = np.mean(phi_m, axis=0)
-    phi_h_mean = np.mean(phi_h, axis=0)
-
-    return {"phi_m": phi_m_mean, "phi_h": phi_h_mean}
+    min_wind_speed: float,
+) -> NDArray[np.float32]:
+    r"""Calculate wind speed profile, [m s-1].
 
+    The wind speed at different heights is calculated using the following equation:
 
-def calculate_mean_mixing_length(
-    canopy_height: NDArray[np.float32],
-    zero_plane_displacement: NDArray[np.float32],
-    roughness_length_momentum: NDArray[np.float32],
-    mixing_length_factor: float,
-) -> NDArray[np.float32]:
-    """Calculate mixing length for canopy air transport, [m].
+    .. math::
+        u(z) = u_{ref} \times \frac{ \ln \left( \frac{z - d}{z_0} \right) }
+                                { \ln \left( \frac{z_{ref} - d}{z_0} \right) }
 
-    The mean mixing length is used to calculate turbulent air transport inside vegetated
-    canopies. It is made equivalent to the above canopy value at the canopy surface. In
-    absence of vegetation, it is set to zero. Implementation after
-    :cite:t:`maclean_microclimc_2021`.
+    where :math:`u(z)` is the wind speed at height :math:`z`, :math:`u_{ref}` is the
+    reference wind speed at reference height :math:`z_{ref}`, :math:`z` is the height at
+    which the wind speed is calculated, :math:`z_0` is the roughness length, and
+    :math:`d` is the zero plane displacement.
 
     Args:
-        canopy_height: Canopy height, [m]
+        reference_wind_speed: Reference wind speed above the canopy, [m s-1].
+        reference_height: Reference height above the canopy, [m].
+        wind_heights: Heights where wind speed is to be calculated, [m].
+        roughness_length: Momentum roughness length, [m]
         zero_plane_displacement: Height above ground within the canopy where the wind
             profile extrapolates to zero, [m]
-        roughness_length_momentum: Momentum roughness length, [m]
-        mixing_length_factor: Factor in calculation of mean mixing length, dimensionless
+        min_wind_speed: Minimum wind speed, [m s-1]
```

Comment:
Can you explain this in the docstring? My assumption is that something breaks horribly if it wind speed hits zero?

Reply:
yes, should not divide by zero